lnwallet/channel.go

```diff
@@ -546,6 +547,13 @@ type LightningChannel struct {
 	// way to lookup the original PaymentDescriptor.
 	rHashMap map[PaymentHash][]*PaymentDescriptor
 
+	// [Our|Their]ExpectedBalance reflects the available balance both locally
+	// and remotely. Every time an update happens such as adding or receiving
+	// a HTLC, these values are updated immediately to reflect the available
+	// balance and make sure they don't go below the channel reserve limit.
+	OurExpectedBalance   btcutil.Amount
```

Comment:
Also with the semantics of the new state machine, updates in the log aren't guaranteed to be included in the next state transition if both sides add to the log concurrently